src/tools/kata-ctl/src/ops/env_ops.rs

```diff
@@ -0,0 +1,468 @@
+        get_command_version(&hypervisor_config.path).context("error getting hypervisor version")?;
+
+    let hypervisor_info = HypervisorInfo {
+        machine_type: hypervisor_config.machine_info.machine_type.to_string(),
+        machine_accelerators: hypervisor_config
+            .machine_info
+            .machine_accelerators
+            .to_string(),
+        version,
+        path: hypervisor_config.path.to_string(),
+        block_device_driver: hypervisor_config
+            .blockdev_info
+            .block_device_driver
+            .to_string(),
+        entropy_source: hypervisor_config.machine_info.entropy_source.to_string(),
+        shared_fs: hypervisor_config
+            .shared_fs
+            .shared_fs
+            .clone()
+            .unwrap_or(String::from("none")),
+        virtio_fs_daemon: hypervisor_config.shared_fs.virtio_fs_daemon.to_string(),
+        msize_9p: hypervisor_config.shared_fs.msize_9p,
+        memory_slots: hypervisor_config.memory_info.memory_slots,
+        pcie_root_port: hypervisor_config.device_info.pcie_root_port,
+        hotplug_vfio_on_rootbus: hypervisor_config.device_info.hotplug_vfio_on_root_bus,
+        debug: hypervisor_config.debug_info.enable_debug,
+        enable_iommu: hypervisor_config.device_info.enable_iommu,
+        enable_iommu_platform: hypervisor_config.device_info.enable_iommu_platform,
+        default_vcpus: hypervisor_config.cpu_info.default_vcpus,
+        cpu_features: hypervisor_config.cpu_info.cpu_features.to_string(),
+    };
+
+    let image_info = ImageInfo {
+        path: hypervisor_config.boot_info.image.clone(),
+    };
+
+    let kernel_info = KernelInfo {
+        path: hypervisor_config.boot_info.kernel.to_string(),
+        parameters: hypervisor_config.boot_info.kernel_params.to_string(),
+    };
+
+    let initrd_info = InitrdInfo {
+        path: hypervisor_config.boot_info.initrd.to_string(),
+    };
+
+    Ok((hypervisor_info, image_info, kernel_info, initrd_info))
+}
+
+pub fn get_env_info(toml_config: &TomlConfig) -> Result<EnvInfo> {
+    let metainfo = get_meta_info();
+
+    let runtime_info = get_runtime_info(toml_config);
+
+    let agent_info = get_agent_info(toml_config)?;
+
+    let host_info = get_host_info()?;
+
+    let (hypervisor_info, _image_info, kernel_info, initrd_info) =
+        get_hypervisor_info(toml_config)?;
+
+    let env_info = EnvInfo {
+        meta: metainfo,
+        runtime: runtime_info,
+        kernel: kernel_info,
+        image: _image_info,
+        initrd: initrd_info,
+        hypervisor: hypervisor_info,
+        host: host_info,
+        agent: agent_info,
+    };
+
+    Ok(env_info)
+}
+
+pub fn handle_env(env_args: EnvArgument) -> Result<()> {
+    let (toml_config, _) = TomlConfig::load_raw_from_file("").context("load toml config")?;
+
+    let env_info = get_env_info(&toml_config)?;
+
+    if env_args.json {
+        let serialized_json = serde_json::to_string_pretty(&env_info).unwrap();
```

Comment:
```suggestion
        let serialized_json = serde_json::to_string_pretty(&env_info)?;
```

Reply:
> Thanks for raising @amshinde !
> 
> A few comments inline. Also, if I run this without `root` perms, I get an error:
> 
> ```shell
> $ kata-ctl env
> ERROR: NoPerms
> $ echo $?
> 1
> ```
> 
> This is coming from `ProtectionError` in `src/check.rs`, but rather than showing `NoPerms`, the error displayed should be:
> 
> ```
> ERROR: No permission to check guest protection
> ```
> 
> From the `#[error]` here: https://github.com/kata-containers/kata-containers/blob/main/src/tools/kata-ctl/src/check.rs#L140..L144.

@jodh-intel Will be looking into this, but I am not sure what I need to be doing differently here. I simple invoke `available_guest_protection` as 
```
let guest_protection: String = arch_specific::available_guest_protection()?.to_string();
```
I would assume that `?` operator would cause the string for the error to return here. Do you know how this should be handled? I'll continue to take a look at this tomorrow.

Reply:
I have fixed this now. I had to explicitly call to_string() on the ProtectionError returned to fix this. I now this this:
```
cargo run -- env 
ERROR: Error {
    context: "get host information",
    source: "No permission to check guest protection",
}
```